lib/plugins/aws/package/compile/functions/index.js

```diff
@@ -441,98 +441,125 @@ class AwsCompileFunctions {
         fileHash.setEncoding('base64');
         versionHash.setEncoding('base64');
 
-        // Read the file in chunks and add them to the hash (saves memory and performance)
-        versionCompilationPromise = BbPromise.fromCallback(cb => {
-          const readStream = fs.createReadStream(artifactFilePath);
-
-          readStream
-            .on('data', chunk => {
-              fileHash.write(chunk);
-              versionHash.write(chunk);
-            })
-            .on('close', () => {
-              cb();
-            })
-            .on('error', error => {
-              cb(error);
+        // Include function and layer configuration in version id hash (without the Code part)
+        versionCompilationPromise = this.addFileContentsToHashes(artifactFilePath, [
+          fileHash,
+          versionHash,
+        ])
+          .then(() => {
+            // Include all referenced layer code in the version id hash
+            const functionProperties = _.omit(_.get(functionResource, 'Properties', {}), 'Code');
```

Comment:
_I know it's direct copy from original, but as we're in process of minimizing lodash usage, it'll be nice to clean this up:_

`functionResource.Properties` is guaranteed to be at this point, therefore let's not use `_.get` here.